systemtest/src/test/java/io/strimzi/systemtest/recovery/NamespaceDeletionRecoveryST.java

```diff
@@ -0,0 +1,255 @@
+            .editSpec()
+                .editKafka()
+                    .withNewPersistentClaimStorage()
+                        .withNewSize("100")
+                        .withStorageClass(storageClassName)
+                    .endPersistentClaimStorage()
+                .endKafka()
+                .editZookeeper()
+                    .withNewPersistentClaimStorage()
+                        .withNewSize("100")
+                        .withStorageClass(storageClassName)
+                    .endPersistentClaimStorage()
+                .endZookeeper()
+            .endSpec().done();
+
+        KafkaClientsResource.deployKafkaClients(false, CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).done();
+
+        String defaultKafkaClientsPodName =
+                ResourceManager.kubeClient().listPodsByPrefixInName(CLUSTER_NAME + "-" + Constants.KAFKA_CLIENTS).get(0).getMetadata().getName();
+
+        internalKafkaClient.setPodName(defaultKafkaClientsPodName);
+
+        LOGGER.info("Checking produced and consumed messages to pod:{}", internalKafkaClient.getPodName());
+        Integer consumed = internalKafkaClient.receiveMessages(topicName, NAMESPACE, CLUSTER_NAME, MESSAGE_COUNT, consumerGroup);
+        assertThat(consumed, is(MESSAGE_COUNT));
+    }
+
+    @Test
+    void testTopicNotAvailable() throws InterruptedException {
+        String topicName = "test-topic-" + new Random().nextInt(Integer.MAX_VALUE);
+
+        prepareEnvironmentForRecovery(topicName, MESSAGE_COUNT);
+
+        // Wait till consumer offset topic is created
+        KafkaTopicUtils.waitForKafkaTopicCreationByNamePrefix("consumer-offsets");
+        // Get list of topics and list of PVC needed for recovery
+        List<PersistentVolumeClaim> persistentVolumeClaimList = kubeClient().getClient().persistentVolumeClaims().list().getItems();
+        deleteAndRecreateNamespace();
+        recreatePvcAndUpdatePv(persistentVolumeClaimList);
+        recreateClusterOperator();
+
+        String consumerGroup = "group-" + new Random().nextInt(Integer.MAX_VALUE);
+        // Recreate Kafka Cluster
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, 3, 3)
+            .editSpec()
+                .editKafka()
+                    .withNewPersistentClaimStorage()
+                        .withNewSize("100")
+                        .withStorageClass(storageClassName)
+                    .endPersistentClaimStorage()
+                .endKafka()
+                .editZookeeper()
+                    .withNewPersistentClaimStorage()
+                        .withNewSize("100")
+                        .withStorageClass(storageClassName)
+                    .endPersistentClaimStorage()
+                .endZookeeper()
+                .withNewEntityOperator()
+                .endEntityOperator()
+            .endSpec().done();
+
+        // Wait some time after kafka is ready before delete topics files
+        Thread.sleep(60000);
```

Comment:
Should you actually check the status or something?

Reply:
This whould be great, but I am not sure what exactly I can check in that case, because Kafka CR is ready, but when I execute the following command to delete topic data from ZK, I got strange errors as I sent you in offline discussion. Only this way help to solve it. If you have any siggestion what I can check instead of this sleep, I am happy to change it